tests/components/application-connector/test/compass-runtime-agent/testkit/director/queryprovider.go

```diff
@@ -0,0 +1,69 @@
+package director
+
+import "fmt"
+
+type queryProvider struct{}
+
+func (qp queryProvider) registerApplicationMutation(appName, _ string) string {
+	return fmt.Sprintf(`mutation {
+	result: registerApplication(in: {
+		name: "%s"
+	}) { id } }`, appName)
+}
+
+func (qp queryProvider) registerApplicationFromTemplateMutation(appName, displayName string) string {
+	return fmt.Sprintf(`mutation {
+	result: registerApplicationFromTemplate(in: {
+		templateName: "SAP Commerce Cloud"
+		values: [
+			{ placeholder: "name", value: "%s" }
+			{ placeholder: "display-name", value: "%s" }
+        ]
+	}) { id } }`, appName, displayName)
+}
+
+func (qp queryProvider) assignFormationForAppMutation(applicationId, formationName string) string {
+	return fmt.Sprintf(`mutation {
+	result: assignFormation(
+		objectID: "%s"
+		objectType: APPLICATION
+		formation: { name: "%s" }
+	) { id } }`, applicationId, formationName)
+}
+
+func (qp queryProvider) assignFormationForRuntimeMutation(runtimeId, formationName string) string {
```

Comment:
Shouldn't we also have some mutations to unassign the formations from Applications and Runtimes?

Reply:
During cleanup it is not necessary. We can just delete Runtime and application. But I think I forgot one thing adding /removing formation. We should discuss it